### ISSUE TITLE
Namespace in sensort definition doesn't seem to work

### ISSUE THREAD

**Issue by wigster** [danieldotnl/ha-multiscrape#114]
## Version of the custom_component
6.2.0

## Configuration

```yaml
- name: Wolfram Sunburn Time
  resource: https://api.wolframalpha.com/v2/query?input=time+to+sunburn+skin+type+i+prague+cz&format=plaintext&output=JSON&appid=[id]
  scan_interval: 1500
  method: GET
  log_response: true
  sensor:
    - unique_id: wolfram_prague_sunburn_time_skin_i
      name: Sunburn Time Prague Skin Type I
      value_template: >
        {%- set bla = value_json.queryresult.pods[1].subpods[0].plaintext.split('|')[5] %}
        {%- set timestring =  bla.split(' ') %}
        {%- set ns = namespace(minutes = 0) %}
        {%- for i in range((timestring | length / 2) | int) %}
        {%- if timestring[i * 2 + 1] == 'h' %}
          {%- set ns.minutes = ns.minutes + (timestring[i * 2] | float) * 60 %}
        {%- else %}
          {%- set ns.minutes = ns.minutes + (timestring[i * 2] | float) %}
        {%- endif %}
        {%- endfor %}
        {{ns.minutes}}
      state_class: measurement
      attributes:
        - name: SPF15
          value_template: "{{value_json.queryresult.pods[1].subpods[0].plaintext.split('|')[6]}}"
        - name: SPF30
          value_template: "{{value_json.queryresult.pods[1].subpods[0].plaintext.split('|')[7]}}"
        - name: SPF45
          value_template: "{{value_json.queryresult.pods[1].subpods[0].plaintext.split('|')[8]}}"
```

## Describe the bug

Multiscrape fetches the data without issues, but the template does not get processed in the way that I would like.

For some reason the value of the ns.minutes variable that is returned by this sensor is always what I set it to in the line where I define ns above the for/if loop (i.e. in this case 0). The output is correct in the Templates Developer Tools. 

Does multiscrape not support a writing into a namespace in a loop in its value_template?

(Just for concreteness, the template is complex, because the server returns the time in one of the formats (`3.5 h`, `1 h 50 min` or `50 min`) and I want to convert it to minutes for the output of the sensor)

## Debug log

```text

Add your logs here.

```

**Comment by danieldotnl** [comment]
Could you help a bit and try if it works when you process the raw result of multiscrape with a template sensors?

**Comment by danieldotnl** [comment]
You removed your comment in which it became clear that this is not a multiscrape issue? :)

**Comment by wigster** [comment]
I wanted to say it the external template didn't work, but then I realised belatedly that the source in the mean time changed to a string saying "not likely", so not within my cases.

Now that it's day again, the source website is giving "40 min", multiscrape scrapes that correctly to a value_json and then the template sensor containing the template with the namespace works correctly to convert that to a float of 40.0.

I noticed that you upgraded to 3.1.2, so I have tried again in the latest release, but the problem with defining this template within multiscrape directly is the same as it was in 3.1.0.
